lib/driver.js

```diff
@@ -244,21 +285,31 @@ class EspressoDriver extends BaseDriver {
       }
     };
 
-    const isApk = _.endsWith(_.toLower(appPath), '.apk');
-    // Do not cache .apk if was already present on the local file system
-    const shouldResultAppPathBeCached = !isApk || (isApk && isUrl);
     let pathInCache = null;
     let isResultAppPathAlreadyCached = false;
     if (_.isPlainObject(cachedAppInfo)) {
       const packageHash = await fs.hash(appPath);
       if (packageHash === cachedAppInfo.packageHash && await fs.exists(cachedAppInfo.fullPath)) {
+        logger.info(`Using '${cachedAppInfo.fullPath}' that is cached from '${appPath}'`);
         isResultAppPathAlreadyCached = true;
         pathInCache = cachedAppInfo.fullPath;
       }
     }
+
+    // appPath can be .zip, .apk, .apks and .aab
+    const isApk = this.isApk(appPath);
+    // Do not cache .apk if was already present on the local file system
+    const shouldResultAppPathBeCached = !isApk || (isApk && isUrl);
```

Comment:
I assume shouldResultAppPathBeCached to be true for resulting .apk extracted from a zip archive